### PR TITLE
[For 3.0.0] Feature/2917 adding username to user at redux

### PR DESCRIPTION
**Pull request by gbonaspetti**
#2917 

## Checkpoints

The goal of this section is to help code reviewer and merge request author to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewer**

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

**For developper**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] Original issue have been updated with a short resume of implemented solution

**For quality team**

- [x] Tested by quality team

### CODE REVIEW

**Comment by gbonaspetti** [comment]
Automated tests have not been written because it is not a visible feature that can be tested with our current infrastructure.

File: frontend/test/src/container/Account.spec.js [hunk 1]
```diff
@@ -115,12 +115,12 @@ describe('<Account />', () => {
     })
 
     describe('handleSubmitNameOrEmail', () => {
-      it('updateUserNameCallBack should be called when the function handleSubmitNameOrEmail() is called with a new Name', (done) => {
+      it('updateUserPublicNameCallBack should be called when the function handleSubmitNameOrEmail() is called with a new Name', (done) => {
```
Comment (Skylsmoi) [inline frontend/test/src/container/Account.spec.js:118]: take advantage of this PR to rename this test. It should start with "should"

Comment (gbonaspetti) [reply frontend/test/src/container/Account.spec.js:118]: ok, I'll rename all the tests in that file to be consistent

File: frontend/test/src/reducer/user.spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  logged: null,
+  auth_type: '',
+  timezone: '',
+  profile: PROFILE.user,
+  email: '',
+  is_active: true,
+  avatar_url: null,
+  created: '',
+  public_name: '',
+  lang: getBrowserLang(),
```
Comment (Skylsmoi) [inline frontend/test/src/reducer/user.spec.js:22]: I don't think you should be using getBrowerLang().
The unit tests aren't run inside a browser, I don't know what to expect from getBrowerLang() but I'm assuming it is undefined or a default value.

File: frontend/test/src/reducer/user.spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import userReducer from '../../../src/reducer/user.js'
+import {
+  UPDATE,
+  USER_USERNAME,
+  USER_PUBLIC_NAME
+} from '../../../src/action-creator.sync'
+import { getBrowserLang } from '../../../src/helper.js'
+import { PROFILE } from 'tracim_frontend_lib'
+
+const defaultUser = {
```
Comment (Skylsmoi) [inline frontend/test/src/reducer/user.spec.js:11]: your test user should have actual values.
There is a user in the fixture folder, you can use it.

Check that this user has all the properties and update it if required

Comment (gbonaspetti) [reply frontend/test/src/reducer/user.spec.js:11]: The purpose of defaultUser was to test that the reducer "should return the default state", it is equal to the default state, so it has no values.
For the other tests I will use the user in the fixture folder, but for this "it" test I still think it is important to keep the defaultUser.

Comment (Skylsmoi) [reply frontend/test/src/reducer/user.spec.js]: I see. Then you need to get the defaultUser variable from the reducer's file

Comment (Skylsmoi) [reply frontend/test/src/reducer/user.spec.js]: export it if it is required

**Comment by metzgegu** [comment]
Tested and verified on branch.

**Pull request merged**